pkgs/development/python-modules/pc-ble-driver-py/default.nix

```diff
@@ -0,0 +1,51 @@
+{ stdenv, buildPythonPackage, fetchpatch, fetchFromGitHub,
+  python, cmake, git, swig, boost, udev,
+  setuptools, enum34, wrapt, future }:
+
+buildPythonPackage rec {
+  pname = "pc-ble-driver-py";
+  version = "0.11.4";
+  disabled = python.isPy3k;
+
+  src = fetchFromGitHub {
+    owner = "NordicSemiconductor";
+    repo = "pc-ble-driver-py";
+    rev = "v${version}";
+    fetchSubmodules = true;
+    sha256 = "0lgmcnrlcivmawmlcwnn4pdp6afdbnf3fyfgq22xzs6v72m9gp81";
+  };
+
+  nativeBuildInputs = [ cmake swig git setuptools ];
+  buildInputs = [ boost udev ];
+  propagatedBuildInputs = [ enum34 wrapt future ];
+
+  patches = [
+    # build system expects case-insensitive file system
+    (fetchpatch {
+      url = "https://patch-diff.githubusercontent.com/raw/NordicSemiconductor/pc-ble-driver-py/pull/84.patch";
+      sha256 = "0ibx5g2bndr5h9sfnx51bk9b62q4jvpdwhxadbnj3da8kvcz13cy";
+    })
+  ];
+
+  postPatch = ''
+    # do not force static linking of boost
+    sed -i /Boost_USE_STATIC_LIBS/d pc-ble-driver/cmake/*.cmake
+
+    cd python
+  '';
+
+  preBuild = ''
+    pushd ../build
+    cmake ..
+    make -j $NIX_BUILD_CORES
```

Comment:
This package is a bit tricky to build.  There is a `python/setup.py`, however the `setup.py` does not build the native extension.  The native extension is instead built by a top-level cmake file that *copies* the binary into the `python` directory.  The pypi packages bundle the pre-built binaries...